src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -0,0 +1,410 @@
+                    break;
+            }
+
+            if (OperationType::ITEM === $operationType) {
+                $responses['404'] = new Model\Response('Resource not found');
+            }
+
+            if (!$responses) {
+                $responses['default'] = new Model\Response('Unexpected error');
+            }
+
+            $requestBody = null;
+            if ('PUT' === $method || 'POST' === $method || 'PATCH' === $method) {
+                /** @TODO support multiple input formats */
+                $operationFormatInput = array_values($requestMimeTypes)[0];
+                $operationSchemaInput = $this->jsonSchemaFactory->buildSchema($resourceClass, $operationFormatInput, Schema::TYPE_INPUT, $operationType, $operationName, new Schema('openapi'), $context);
+                $schemas += $operationSchemaInput->getDefinitions()->getArrayCopy();
+                $requestBody = new Model\RequestBody(sprintf('The %s %s resource', 'POST' === $method ? 'new' : 'updated', $resourceShortName), $this->buildContent($requestMimeTypes, $operationSchemaInput), true);
+            }
+
+            $pathItem = $pathItem->{'with'.ucfirst($method)}(new Model\Operation(
+                $operationId,
+                $operation['openapi_context']['tags'] ?? [$resourceShortName],
+                $responses,
+                $operation['openapi_context']['summary'] ?? '',
+                $operation['openapi_context']['description'] ?? $this->getPathDescription($resourceShortName, $method, $operationType),
+                isset($operation['openapi_context']['externalDocs']) ? new ExternalDocumentation($operation['openapi_context']['externalDocs']['description'] ?? null, $operation['openapi_context']['externalDocs']['url']) : null,
+                $parameters,
+                $requestBody,
+                isset($operation['openapi_context']['callbacks']) ? new \ArrayObject($operation['openapi_context']['callbacks']) : null,
+                $operation['openapi_context']['deprecated'] ?? (bool) $resourceMetadata->getTypedOperationAttribute($operationType, $operationName, 'deprecation_reason', false, true),
+                $operation['openapi_context']['security'] ?? [],
+                $operation['openapi_context']['servers'] ?? []
+            ));
+
+            $paths->addPath($path, $pathItem);
+        }
+
+        return [$links, $schemas];
+    }
+
+    /**
+     * @TODO: support multiple format schemas
+     */
+    private function buildContent(array $responseMimeTypes, $operationSchema): \ArrayObject
+    {
+        $content = new \ArrayObject();
+
+        foreach ($responseMimeTypes as $mimeType => $format) {
+            $content[$mimeType] = new Model\MediaType(new \ArrayObject($operationSchema->getArrayCopy(false)));
+        }
+
+        return $content;
+    }
+
+    private function getMimeTypes($resourceClass, $operationName, $operationType, $resourceMetadata = null)
```

Comment:
```suggestion
    private function getMimeTypes($resourceClass, $operationName, $operationType, $resourceMetadata = null): array
```

Reply:
typehints for parameters are missing too